src/_includes/macros/form.html

```diff
@@ -0,0 +1,244 @@
+{# ==========================================================================
+
+   form.input()
+
+   ==========================================================================
+
+   Description:
+
+   Builds form text field.
+
+   label: Name of the field.
+
+   type: Type of field. Will default to text, but can change to other
+         types to take advantage of HTML5 validation
+          Possible fields: text, tel, email, url, number, date.
+
+   size: The size of the field. Defaults to full-width.
+          Possible options: 1, 1-2, 1-3
+
+   required: Whether the field is required. Defaults to false.
+
+   disabled: Whether the field is disabled. Defaults to false.
+
+   ========================================================================== #}
+
+{% macro input(label, type='text', size='1', required=false, disabled=false) -%}
+  {% from 'macros/util/format/url.html' import slugify as slugify %}
```

Comment:
Stuck with 4 indents throughout. I was under the impression jinja was two spaces, but I'm not going to start a new developer war over the level of indentation. :smile: